server/src/instant/db/transaction.clj

```diff
@@ -349,7 +349,13 @@
                           (triple-model/delete-multi! conn app-id (map next tx-steps))))))
              {}
              grouped-tx-steps)
-
+            eid+attr-ids (distinct
+                          (concat
+                           (:delete-entity results)
+                           (:add-triple results)
+                           (:deep-merge-triple results)
+                           (:retract-triple results)))
+            _  (triple-model/validate-required! conn app-id eid+attr-ids)
```

Comment:
I'm worried about the extra queries that we're adding to `transact`. 

We have the attrs here, could we at least skip this query if there are no entities with required attrs that are affected by this transaction?